### PR TITLE
Fix Lang Server issues on Hover and Completions

### LINKED ISSUE #26044: Fork statement should not suggest within `isolated` functions
**Description:**
$title

**Affected Versions:**
slp4

**OS, DB, other environment details and versions:**

**Related Issues (optional):**
#26042

### LINKED ISSUE #29392: Invalid completions at the module level
**Description:**
Consider the following source snippet

```
import ballerina/io;

public function main() {
    io:println("Hello World!");
}
<cursor>

```
At the given cursor position, we get  only the `returns` keyword as the suggestion

**Affected Versions:**
sl-alpha3 at least

### LINKED ISSUE #29450: Hovering over types does not show anything
**Description:**

eg: hovering over `http:Response`
```ballerina
import ballerina/io;
import ballerina/http;

public string API_KEY = "";
http:Client endpoint = check new("https://free.currconv.com/api/v7/convert");

function rate(Currency 'from, Currency to) {
    io:println("Requesting rates from "+ <string>'from + " to " + <string>to);

    http:Response|http:ClientError response = endpoint->get(getParamUrlString('from, to));

    if response is http:Response {
        io:println(response.toString());
    }
    
}
```

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**
SL Alpha 3

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

### LINKED ISSUE #29455: Completions are not provided for action invocation
**Description:**
Consider the following source snippet and $title at the cursor.

```
import ballerina/websocket as ws;

class FinnHubListener {
    private ws:Client wsClient;
    private decimal pollInterval;
    private boolean proceed = false;

    function init(ws:Client wsClient, decimal pollInterval) {
        self.wsClient = wsClient;
        self.pollInterval = pollInterval;
    }

    public function poll(string symbolName) {
        string subscriptionMsg = string `"type":"subscribe","symbol":"${symbolName}"}`;
        ws:Error? resp = self.wsClient->writeTextMessage(subscriptionMsg);
        if resp is ws:Error {
            return;
        }
        self.wsClient-><cursor>
        self.proceed = true;
        while self.proceed {

            io:println("Poll");
            runtime:sleep(self.pollInterval);
        }
    }

    public function stop() {
        io:println("interrupted!");
        self.proceed = false;
    }
}
```

**Affected Versions:**
sl-alpha3 at least

### LINKED ISSUE #29561: Quote should be added when importing a module which is a reserved keyword
**Description:**
Consider the following sample.

When giving suggestion/completion quote should to added to the module name if the module name is a reserved keyword. Otherwise, completion will be invalid and can not be parsed by the parser. 

e.g. Here completion should be `import lcPortal.'client;`

Screenshot from 2021-03-24 09-54-13

### PR DESCRIPTION
**Pull request by nadeeshaan**
## Purpose
Fixes #29561 
Fixes #29455 
Fixes #29450
Fixes #29392 
Fixes #26044 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/RemoteMethodCallActionNodeContext.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/RemoteMethodCallActionNodeContext.java]: Year should be 2021.

Comment (dulajdilshan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/RemoteMethodCallActionNodeContext.java:2]:
```suggestion
 * Copyright (c) 2021, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```

**Review by dulajdilshan — APPROVED** [review]
LGTM

**Review by IMS94 — APPROVED** [review]
LGTM! Need to update a copyright header's year before merging.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/RemoteMethodCallActionNodeContext.java (hunk 1 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/RemoteMethodCallActionNodeContext.java]: Disregard this as it has been resolved with a previous comment.

**Review by mohanvive — APPROVED** [review]

**Pull request merged**